src/main/java/com/cloudant/client/api/Database.java

```diff
@@ -64,6 +64,8 @@
 /**
  * Contains a Database Public API implementation.
  *
+ * Methods may throw a {@link NoDocumentException} if the database does not exist.
```

Comment:
Sadly, yes. The `CouchDbException` hierarchy are all descended from `RuntimeException`. I think we'd like to address that at some point in the future, but it would be a big breaking change that I don't think we're ready for yet.